amy/emails/actions/instructor_training_approaching.py

```diff
@@ -0,0 +1,228 @@
+    return result
+
+
+def run_instructor_training_approaching_strategy(
+    strategy: StrategyEnum, request: HttpRequest, event: Event
+) -> None:
+    mapping = {
+        StrategyEnum.CREATE: instructor_training_approaching_signal,
+        StrategyEnum.UPDATE: instructor_training_approaching_update_signal,
+        StrategyEnum.REMOVE: instructor_training_approaching_remove_signal,
+    }
+    if strategy not in mapping:
+        return
+
+    signal = mapping[strategy]
+
+    logger.debug(f"Sending signal for {event} as result of strategy {strategy}")
+    signal.send(
+        sender=event,
+        request=request,
+        event=event,
+        event_start_date=event.start,
+    )
+
+
+@receiver(instructor_training_approaching_signal)
+@feature_flag_enabled("EMAIL_MODULE")
+def instructor_training_approaching_receiver(
+    sender: Any, **kwargs: Unpack[InstructorTrainingApproachingKwargs]
+) -> None:
+    request = kwargs["request"]
+    event = kwargs["event"]
+    event_start_date = kwargs["event_start_date"]
+    instructors = [
+        task.person
+        for task in Task.objects.filter(event=event, role__name="instructor")
+    ]
+    instructor_emails = [
+        instructor.email for instructor in instructors if instructor.email
+    ]
+
+    scheduled_at = one_month_before(event_start_date)
+    context: InstructorTrainingApproachingContext = {
+        "event": event,
+        "instructors": instructors,
+    }
+    signal = instructor_training_approaching_signal.signal_name
+    try:
+        scheduled_email = EmailController.schedule_email(
+            signal=signal,
+            context=context,
+            scheduled_at=scheduled_at,
+            to_header=instructor_emails,
+            generic_relation_obj=event,
+            author=person_from_request(request),
+        )
+    except EmailControllerMissingRecipientsException:
+        messages_missing_recipients(request, signal)
+    except EmailTemplate.DoesNotExist:
+        messages_missing_template(request, signal)
+    else:
+        messages_action_scheduled(request, signal, scheduled_email)
+
+
+@receiver(instructor_training_approaching_update_signal)
+@feature_flag_enabled("EMAIL_MODULE")
+def instructor_training_approaching_update_receiver(
+    sender: Any, **kwargs: Unpack[InstructorTrainingApproachingKwargs]
+) -> None:
+    request = kwargs["request"]
+    event = kwargs["event"]
+    event_start_date = kwargs["event_start_date"]
+    instructors = [
+        task.person
+        for task in Task.objects.filter(event=event, role__name="instructor")
+    ]
+    instructor_emails = [
+        instructor.email for instructor in instructors if instructor.email
+    ]
+
+    scheduled_at = one_month_before(event_start_date)
+    context: InstructorTrainingApproachingContext = {
+        "event": event,
+        "instructors": instructors,
+    }
+    signal = instructor_training_approaching_signal.signal_name
```

Comment:
I'm making some changes to hopefully capture this better. So all these 3 signals use the same string value for their signal name.

I'm going to switch to a constant value `INSTRUCTOR_TRAINING_APPROACHING_SIGNAL_NAME = "instructor_training_approaching"` so that it's used instead of `.signal_name` property.